bindings/matrix-sdk-ffi/src/platform.rs

```diff
@@ -245,12 +231,66 @@ pub struct TracingFileConfiguration {
     max_files: Option<u64>,
 }
 
+#[derive(PartialEq, PartialOrd)]
+enum LogTarget {
+    Hyper,
+    MatrixSDKFFI,
+    MatrixSDKClient,
+    MatrixSDKCrypto,
+    MatrixSDKCryptoAccount,
+    MatrixSDKOIDC,
+    MatrixSDKHTTPClient,
+    MatrixSDKSlidingSync,
+    MatrixSDKBaseSlidingSync,
+    MatrixSDKUITimeline,
+    MatrixSDKEventCache,
+    MatrixSDKEventCacheStore,
+}
+
+impl LogTarget {
+    fn as_str(&self) -> &'static str {
+        match self {
+            LogTarget::Hyper => "hyper",
+            LogTarget::MatrixSDKFFI => "matrix_sdk_ffi",
+            LogTarget::MatrixSDKClient => "matrix_sdk::client",
+            LogTarget::MatrixSDKCrypto => "matrix_sdk_crypto",
+            LogTarget::MatrixSDKCryptoAccount => "matrix_sdk_crypto::olm::account",
+            LogTarget::MatrixSDKOIDC => "matrix_sdk::oidc",
+            LogTarget::MatrixSDKHTTPClient => "matrix_sdk::http_client",
+            LogTarget::MatrixSDKSlidingSync => "matrix_sdk::sliding_sync",
+            LogTarget::MatrixSDKBaseSlidingSync => "matrix_sdk_base::sliding_sync",
+            LogTarget::MatrixSDKUITimeline => "matrix_sdk_ui::timeline",
+            LogTarget::MatrixSDKEventCache => "matrix_sdk::event_cache",
+            LogTarget::MatrixSDKEventCacheStore => "matrix_sdk_sqlite::event_cache_store",
+        }
+    }
+}
+
+const DEFAULT_TARGET_LOG_LEVELS: &[(LogTarget, LogLevel)] = &[
+    (LogTarget::Hyper, LogLevel::Warn),
+    (LogTarget::MatrixSDKFFI, LogLevel::Info),
+    (LogTarget::MatrixSDKClient, LogLevel::Trace),
+    (LogTarget::MatrixSDKCrypto, LogLevel::Debug),
+    (LogTarget::MatrixSDKCryptoAccount, LogLevel::Trace),
+    (LogTarget::MatrixSDKOIDC, LogLevel::Trace),
+    (LogTarget::MatrixSDKHTTPClient, LogLevel::Debug),
+    (LogTarget::MatrixSDKSlidingSync, LogLevel::Info),
+    (LogTarget::MatrixSDKBaseSlidingSync, LogLevel::Info),
+    (LogTarget::MatrixSDKUITimeline, LogLevel::Info),
+    (LogTarget::MatrixSDKEventCache, LogLevel::Info),
+    (LogTarget::MatrixSDKEventCacheStore, LogLevel::Info),
+];
+
+const IMMUTABLE_TARGET_LOG_LEVELS: &[LogTarget] = &[LogTarget::Hyper];
+
 #[derive(uniffi::Record)]
 pub struct TracingConfiguration {
-    /// A filter line following the [RUST_LOG format].
-    ///
-    /// [RUST_LOG format]: https://rust-lang-nursery.github.io/rust-cookbook/development_tools/debugging/config_log.html
-    filter: String,
+    /// The desired LogLevel, corresponding to https://docs.rs/log/latest/log/enum.Level.html
```

Comment:
No need to indicate the type, since docs would include a link to the field's type.

```suggestion
    /// The desired log level.
```